lib/compiler.js

```diff
@@ -634,10 +665,26 @@ function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
       _.extend(meta, require(metafile));
     }
     meta.sourceFile = filepath;
-    return meta;
+    mixinInstructions[meta.name] = meta;
   });
 
-  return mixins;
+  return _.values(mixinInstructions);
+}
+
+function fetchMixinNamesUsedInModelInstructions(modelInstructions) {
+  return _.flatten(modelInstructions
+  .map(function(model) {
+    return model.definition && model.definition.mixins ?
+      Object.keys(model.definition.mixins) : [];
+  }));
+}
+
+function filterMixinInstructionsUsingWhitelist(mixinInstructions,
+  includeMixins) {
```

Comment:
Perhaps `(instructions, names)` would fit into 80 chars? If not, then please indent the second line by two stops to provide at least some sort of visual distinction between the arg list and the function body.

```js
function(....
    ...) {
  // code
}
```